graphql/resolvers.go

```diff
@@ -1502,37 +1502,31 @@ func (r *queryResolver) Projects(ctx context.Context) ([]*GroupedProjects, error
 		return nil, ResourceNotFound.Send(ctx, err.Error())
 	}
 
-	groupsMap := make(map[string][]*restModel.APIProjectRef)
-
-	for _, p := range allProjs {
-		groupName := strings.Join([]string{p.Owner, p.Repo}, "/")
-		apiProjectRef := restModel.APIProjectRef{}
-		if err = apiProjectRef.BuildFromService(p); err != nil {
-			return nil, InternalServerError.Send(ctx, fmt.Sprintf("error building APIProjectRef from service: %s", err.Error()))
-		}
-
-		if projs, ok := groupsMap[groupName]; ok {
-			groupsMap[groupName] = append(projs, &apiProjectRef)
-		} else {
-			groupsMap[groupName] = []*restModel.APIProjectRef{&apiProjectRef}
-		}
+	groupedProjects, err := GroupProjects(allProjs, false)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("error grouping project: %s", err.Error()))
 	}
+	return groupedProjects, nil
+}
 
-	groupsArr := []*GroupedProjects{}
+func (r *queryResolver) ViewableProjects(ctx context.Context) ([]*GroupedProjects, error) {
```

Comment:
I don't think a separate resolver is necessary these seem to be doing the same thing. Your version of the resolver actually seems more accurate  since its checking a users permissions and identifying what projects they have permission to view. Could we add the permission check to the projects resolver above and consolidate the two?

Reply:
My resolver only groups projects by repo if they are defaulted to the repo, whereas the old one groups all of them, whether they are defaulted or not.

Reply:
after a conversation with Sophie, I realized that we shouldn't be fetching the entire project Ref here, so this won't be relevant anymore (unless that's also the case where you are using this)

Reply:
Can this also be ViewableProjectSettings ?

Reply:
Since the resolver "Projects" returns projects, I'm worried that people will expected "ViewableProjectSettings" to return ProjectSettings (as opposed to projects).

Reply:
hmm I see your point. Maybe ViewableProjectRef?  "Project" is just a bit vague.

Reply:
That sounds good, I'll update it to ViewableProjectRefs